### DISCUSSION TITLE
i want to know how to use custom 3d models

### DISCUSSION THREAD

**Discussion by hacker101red** [pokepetter/ursina#254] [Q&A] [answered]
can anyone help me with how to use custom 3d models

**Comment by PixelTomato** [comment, marked as answer]
To create an entity with a custom model, you should be able to do something like this:

```py
from ursina import *

app = Ursina()

custom_model = load_model("name_of_model_file.obj")
Entity(model=custom_model)

app.run()
```

Make sure to replace "name_of_model_file.obj" with the actual path of your custom model relative to your project's directory and you should see your custom model appear on the screen when you run your script.

**Reply by hacker101red** [reply]
thx

**Reply by kowshik1234** [reply]
I want to also load the color data of the model? I am only importing the outlines even though I have the mtl file. How can I do that?

Here are my 3D model files. 

[3D_model.zip](https://github.com/pokepetter/ursina/files/13206716/3D_model.zip)

**Reply by pokepetter** [reply]
This is the correct way: https://www.ursinaengine.org/entity_basics.html#Model
```py
Entity(model='name_of_your_model')
```

**Reply by kowshik1234** [reply]
@pokepetter  It didn't work. The below attached is my blender file and when I imported it the render is as below

<img width="291" alt="image" src="https://github.com/pokepetter/ursina/assets/26191169/ff96f525-a774-45e4-861e-72bb74a971ff">

but the render is expected to be as below

<img width="227" alt="image" src="https://github.com/pokepetter/ursina/assets/26191169/3f4d53ad-656b-479c-9b80-4dd8648a2926">

I tried importing the blender, OBJ files but the render is incorrect. Please advise what is the best way.

Thanks

**Reply by pokepetter** [reply]
I'm sorry, my answer was a correction to this answer, not an answer to yours.

Your model doesn't work because you didn't triangualte the mesh. There's a toggle for this on export.

**Reply by kowshik1234** [reply]
@pokepetter How do I correct my model please?

**Comment by hacker101red** [comment]
@PixelTomato   i asked a new question can you please answer them?

**Reply by PixelTomato** [reply]
We need to know what the issue is before we can help.

**Reply by hacker101red** [reply]
the issue is it is always showing game over and i can't see that ground2 and the spikes
and that is in a new question
